Update core dependency and adapt the code

dmitryax opened the pull request:
- `pmetric.MetricValueType` -> `pmetric.NumberDataPointValueType` and `pmetric.ExemplarValueType`

Deprecated usage of `configunmarshaler` will be handled separately

mx-psi reviewed (dismissed):
Same as with the other huge PR: I had a look at a few files at random and they look good, but I haven't checked all 566 files for my own sanity

bogdandrutu commented:
Please rebase and fix lint.

dmitryax committed:
Update core dependency and adopt the code

- pmetric.MetricValueType -> pmetric.NumberDataPointValueType and pmetric.ExemplarValueType
